pallet/dex/src/lib.rs

```diff
@@ -660,19 +659,24 @@ pub mod pallet {
 				Error::<T>::ExpirationMustBeInFuture
 			);
 
-			match order_type {
-				OrderType::SELL => {
-					ensure!(
-						offered_amount * price == requested_amount,
-						Error::<T>::PriceDoNotMatchOfferedRequestedAmount
-					);
-				}
-				OrderType::BUY => {
-					ensure!(
-						offered_amount == requested_amount * price,
-						Error::<T>::PriceDoNotMatchOfferedRequestedAmount
-					);
-				}
+			let (a, b) = match order_type {
+				OrderType::SELL => (requested_amount, offered_amount),
+				OrderType::BUY => (offered_amount, requested_amount),
+			};
+
+			// because price is an integer, we need to check if the division is exact
+			// (does not have a remainder)
+			let price = a
+				.checked_div(&b)
+				.ok_or(Error::<T>::PriceDoNotMatchOfferedRequestedAmount)?;
```

Comment:
This error is division by zero, but up-to-you if you want to fix it.

For some reason, I thought rust had checked_div_mod. Not sure, where I have seen it...

Reply:
![image](https://github.com/ggxchain/ggxnode/assets/1867551/291918c5-96a8-421d-9be2-6e4702cf13cd)
(my screenshot tool went crazy, don't ask why the screenshot looks like this...)